File: need_module/src_test/need_publisher_for_behavior_inter.cpp
#include "ros/ros.h"
#include "social_msg/need_msg.h"

ros::Publisher need_pub;



int main(int argc, char **argv)
{
        ros::init(argc, argv, "need_model_dynamic_reconfigure");
        ros::NodeHandle n;
        // 感知信息
        need_pub  = n.advertise<social_msg::need_msg>("/need_lists", 10);

        sleep(1);
        social_msg::need_msg msg;
        msg.need_name   = config.need_name;          // 需求类型
        msg.scene   = config.need_scene;                // 场景名称 (school, hospital, family)
        msg.person_name   = config.need_person_name;          // 用户姓名
        msg.IDtype   = config.need_IDtype;               // 用户身份
        msg.target_angle   = config.need_target_angle;        // 交互对象角度
        msg.target_distance   = config.need_target_distance;     // 交互对象距离
        msg.rob_emotion   = config.need_rob_emotion;         // 机器人情绪类型
        msg.rob_emotion_intensity   = config.need_rob_emotion_intensity;  // 机器人情绪强度
        msg.person_emotion   = config.need_person_emotion;       // 用户的心情
        msg.weight   = config.need_weight;              // 需求权重
        msg.speech   = config.need_speech;               // 回答文本[机器人回复的话]
        msg.qt_order   = config.need_qt_order ;             // qt显示界面中的索引
        msg.satisfy_value   = config.need_satisfy_value;          // 需求自我满足值[用于情感模块中的自我评价]
        msg.attitude   = config.need_attitude;             // (机器人对用户的)社交态度
        msg.move_speed   = config.need_move_speed;          // 移速
        msg.distance   = config.need_distance;            // 社交距离
        msg.voice_speed   = config.need_voice_speed;         // 语速

        need_pub.publish(msg);



        ros::spin();
        return 0;
}

File: need_module/script/need_publisher_for_behavior_inter.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

from re import I
import numpy as np
import math
from bisect import bisect_left
import pandas as pd
import csv
import rospy
import time
import threading
from social_msg.msg import attitude_msg
from social_msg.msg import perception_msg
from social_msg.msg import robot_emotion
from social_msg.msg import need_msg
import sys
import os

# "/home/zhjd/ws/src/social_system/emotion_module/scripts/"+'emotion_param.csv'
# /home/zhjd/ws/src/social_system/emotion_module/scripts/"+'emotion_param.csv

# csv_name = "/home/zhjd/ws/src/social_system/need_module/script/"+'test(school).csv'
# csv_name = "/home/zhjd/ws/src/social_system/need_module/script/"+'test(school).csv'
sys.stdout.write( '选择刺激输入源,【1】行为打断,【2】行为并行:  ' )
str = input();
current_folder_path = os.path.dirname(os.path.abspath(__file__))




def talker():
        rospy.init_node('need_publisher_for_behavior_inter', anonymous=True)
        pub_need = rospy.Publisher('/need_lists', need_msg, queue_size=10)
        # rate = rospy.Rate(5)
        rate = rospy.Rate(0.08) # 10s发一次
        
        print(" ")
        print(" ")

        if str == 1 :
                time.sleep(1)     
                need = need_msg()

                print("发送对小明的测温需求")
                need.need_name = "MeasureTemperature"  
                need.scene = "school"
                need.person_name = "小明"
                need.IDtype = "学生"
                need.target_angle = 0
                need.target_distance = 1000
                need.rob_emotion = "Joy"
                need.rob_emotion_intensity = 1
                need.person_emotion = "Happy"   
                need.weight = 0.75
                need.speech = ""
                need.qt_order = 0
                need.satisfy_value = 2
                need.attitude = "热情"
                need.move_speed = 200
                need.distance = 700
                need.voice_speed = 50
                pub_need.publish(need)


                time.sleep(15)     #TODO:   具体间隔时间待定

                
                print("发送对小明的维持秩序需求")
                need.need_name = "KeepOrder"
                need.scene = "school"
                need.person_name = "小明"
                need.IDtype = "学生"
                need.target_angle = 0
                need.target_distance = 700
                need.rob_emotion = "Anger"
                need.rob_emotion_intensity = 1
                need.person_emotion = "Happy"   
                need.weight = 0.85 
                need.speech = "请站在原地不要动，小明。我没法测体温了"
                need.qt_order = 0
                need.satisfy_value = 2
                need.attitude = "热情"
                need.move_speed = 200
                need.distance = 1000
                need.voice_speed = 50
                pub_need.publish(need)


        

        if str == 2 :
                time.sleep(1)     
                need = need_msg()


                print("发送对小明的测温需求")
                need.need_name = "MeasureTemperature"  
                need.scene = "school"
                need.person_name = "小明"
                need.IDtype = "学生"
                need.target_angle = 0
                need.target_distance = 1000
                need.rob_emotion = "Joy"
                need.rob_emotion_intensity = 1
                need.person_emotion = "Happy"   
                need.weight = 0.75
                need.speech = ""
                need.qt_order = 0
                need.satisfy_value = 2
                need.attitude = "热情"
                need.move_speed = 200
                need.distance = 700
                need.voice_speed = 50
                pub_need.publish(need)


                time.sleep(15)     #TODO:   具体间隔时间待定

                
                print("发送对李老师的问好需求")
                need.need_name = "RespectGreet"
                need.scene = "school"
                need.person_name = "李老师"
                need.IDtype = "老师"
                need.target_angle = 30
                need.target_distance = 500
                need.rob_emotion = "Trust"
                need.rob_emotion_intensity = 1
                need.person_emotion = "Happy"   
                need.weight = 0.85 
                need.speech = "李老师早上好啊"
                need.qt_order = 0
                need.satisfy_value = 2
                need.attitude = "热情"
                need.move_speed = 200
                need.distance = 500
                need.voice_speed = 50
                pub_need.publish(need)

                        
                        
if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass




